packages/encoding/src/bech32.ts

```diff
@@ -1,15 +1,15 @@
-import * as bech32 from "bech32";
+import { bech32 } from "@scure/base";
 
 export function toBech32(prefix: string, data: Uint8Array, limit?: number): string {
-  const address = bech32.encode(prefix, bech32.toWords(data), limit);
+  const address: `${typeof prefix}1${string}` = bech32.encode(prefix, bech32.toWords(data), limit);
   return address;
 }
 
 export function fromBech32(
   address: string,
   limit = Infinity,
 ): { readonly prefix: string; readonly data: Uint8Array } {
-  const decodedAddress = bech32.decode(address, limit);
+  const decodedAddress = bech32.decode(address as `${string}1${string}`, limit);
```

Comment:
This is an unsafe cast, isn't it? `address` might be any string but we tell `bech32.decode` it is guaranteed to have a certain structure. I guess if we add an

```ts
function isBech32Structure(input: string): input is ${string}1${string} {
    // return false if malformatted
}
```

we get a safe type narrowing